### PR TITLE
Feature: Connect API

### PR DESCRIPTION
**Pull request by Hannziegel**
## In this feature branch we:

> - [x] Used fetch.
> - [x] Used Thunk.
> - [x] Refactored add and remove book features.
> - [x] Add function to get the list of books from the API.
> - [x] No styling added.

### Thank You

### CODE REVIEW

File: src/redux/configureStore.js [hunk 1]
```diff
@@ -1,7 +1,10 @@
-import { configureStore } from '@reduxjs/toolkit';
+import { applyMiddleware, configureStore } from '@reduxjs/toolkit';
+import thunk from 'redux-thunk';
 import bookReducer from './books/books';
 import categoryReducer from './categories/categories';
 
-const store = configureStore({ reducer: { books: bookReducer, categories: categoryReducer } });
+const store = configureStore({
```
Comment (rloterh) [inline src/redux/configureStore.js:6]: - [x] This module requires that you use `createAsyncThunk` for creating _thunk action creators_, which should be similar to this example:

```javascript
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios'; 


export const getBooks = createAsyncThunk(
  'books/getBooks',
  async () => {
    const response = await axios.get(
      'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/',
    );
    return response.data;
  },
); 
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Nunes, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Nice job for using axios for API calls


## Required Changes ♻️
_Kindly check the comments under the review._


<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/configureStore.js (hunk 1 above)
Comment (Hannziegel) [reply src/redux/configureStore.js:6]: Thank you, already done it ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Nunes,  

Your project is complete!  There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉🌟 

## Highlights 👍
- thunk action creators properly implemented
- API correctly connected using Redux

<br>
Keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**